content/en/guide/v10/typescript.md

```diff
@@ -144,47 +141,43 @@ type ExpandableState = {
 };
 
 
-// Bind generics to ExpandableProps and ExpandableState
+// ExpandableProps and ExpandableState are passed as generic type variables.
 class Expandable extends Component<ExpandableProps, ExpandableState> {
-  constructor() {
-    super();
-    // this.state is an object with a boolean field `toggle`
-    // due to ExpandableState
+  constructor(props: ExpandableProps) {
+    super(props);
+    // `this.state` is a object with `toggle` property which is boolean type by ExpandableState.
     this.state = {
       toggled: false
     };
   }
-  // `this.props.title` is string due to ExpandableProps
   render() {
     return (
       <div class="expandable">
         <h2>
+          // `this.props.title` is string type by ExpandableProps.
           {this.props.title}{" "}
-          <button
-            onClick={() => this.setState({ toggled: !this.state.toggled })}
-          >
+          <button onClick={() => this.setState({ toggled: !this.state.toggled })}>
             Toggle
           </button>
         </h2>
+        // `this.props.children` is `ComponentChildren` type by default.
         <div hidden={this.state.toggled}>{this.props.children}</div>
       </div>
     );
   }
 }
 ```
 
-Class components include children by default, typed as `ComponentChildren`.
-
 ## Typing events
 
-Preact emits regular DOM events. As long as your TypeScript project includes the `dom` library (set it in `tsconfig.json`), you have access to all event types that are available in your current configuration.
+Preact handles standard DOM events. When [lib](https://www.typescriptlang.org/tsconfig#lib) of TypeScript compiler options includes `DOM`, you can use standard DOM events types.
 
 ```tsx
 export class Button extends Component {
   handleClick(event: MouseEvent) {
     event.preventDefault();
     if (event.target instanceof HTMLElement) {
-      alert(event.target.tagName); // Alerts BUTTON
```

Comment:
Another one uses `console.log()`.
It is inconsistent.

Reply:
Good call!